### ISSUE TITLE
storage: panic in evalRangeLookup

### ISSUE THREAD

**Issue by couchand** [cockroachdb/cockroach#25360]
https://sentry.io/cockroach-labs/cockroachdb/issues/551645087/

```
*log.safeError: replica_command.go:1374: string
  File "github.com/cockroachdb/cockroach/pkg/storage/replica_command.go", line 199, in evaluateCommand
  File "github.com/cockroachdb/cockroach/pkg/storage/replica.go", line 5062, in evaluateBatch
  File "github.com/cockroachdb/cockroach/pkg/storage/replica.go", line 2458, in executeReadOnlyBatch
  File "github.com/cockroachdb/cockroach/pkg/storage/replica.go", line 1794, in Send
  File "github.com/cockroachdb/cockroach/pkg/storage/store.go", line 2561, in Send
...
(7 additional frame(s) were not displayed)

replica_command.go:1374: string
```

**Comment by couchand** [comment]
Note that this occurred on v1.1.3.  The code around the panic indicates this may be related to #16266, assigning to @nvanbenschoten since he closed that issue.

**Comment by nvanbenschoten** [comment]
This is an assertion failure due to a meta2 split. Splitting the meta2 range was [disabled](https://github.com/cockroachdb/cockroach/pull/17565) in 1.1 and [re-enabled](https://github.com/cockroachdb/cockroach/pull/18970) (while fixing the cause of this assertion) in 2.0. This split must have happened in 1.0 and then the cluster must have been updated to 1.1.3. There isn't anything we can do here other than backporting https://github.com/cockroachdb/cockroach/pull/17565 all the way back to 1.0, but that won't actually fix the issue in clusters that have a meta2 split already. @bdarnell are we planning on doing any more 1.0 releases?

**Comment by bdarnell** [comment]
We'll do more 1.0 releases for supported customers, although I don't think we have any such customers still on 1.0.

Can clusters that have this split recover by upgrading to 2.0, or are they permanently broken?

**Comment by nvanbenschoten** [comment]
> Can clusters that have this split recover by upgrading to 2.0, or are they permanently broken?

Theoretically, yes. 2.0 made range descriptor lookups smarter without placing any new restrictions on where meta2 can split, so it behave fine with a meta2 split created in 1.0.

We don't have a direct upgrade path between 1.0 and 2.0 though, so anyone doing the upgrade will need to first upgrade to 1.1. Depending on where the split is in meta2, this may be a problem for SQL migrations. However, if meta2 is split somewhat evenly, the split will almost certainly be somewhere between descriptors of userspace ranges, so the migration should be able to succeed.

**Comment by nvanbenschoten** [comment]
Talked to @bdarnell about this. Closing since this is an issue in 1.0 that we've known about for a year, it's probably solved by upgrading to 2.0, and there's only been 1 recorded event.
